Add installation request for missing dependencies (bunq/tinker_php#18)

Crecket opened the pull request:
## This PR closes/fixes the following issues:
 - Closes bunq/tinker_php#18
    - [x] Tested

Crecket committed:
Add installation request for missing dependencies (bunq/tinker_php#18)

OGKevin commented on setup.sh:
```diff
@@ -129,7 +145,17 @@ function determineInstructionInstallation
+    warningMessage=$(printf "${ERROR_COULD_NOT_FIND_COMMAND}" "${programName}" "${!commandInstallationConstantName} ${programPackageName}")
+
+    echo ${warningMessage} >$(tty)
+    echo "Do you want to try and install this now? (y/N)? " >$(tty)
```
Please move this string into a consant.

OGKevin commented on setup.sh:
```diff
@@ -57,7 +57,23 @@ function assertAllPrerequisitePresent
+{
+    phpModules=$(php -m | head)
+
+    if [[ ${phpModules} != *"curl"* ]]; then
```
There are some extension missing according to: [composer.json](https://github.com/bunq/sdk_php/blob/8c1faefc111d9b970168a1837ca165d854954941/composer.json)

Aslo this seems like code repeation. Please create a new method:

1. isModlueMissing(string[] $allModules, string $moduleToFind): bool
 You can do this via:

```
allModule=$(php -m)
allNeededModule=(some smart way to get a string[] of everthig in composer json)


for line in $allNeededModule; do
  if (isModlueMissing($allModule $line)); then
      determineInstructionInstallation ...
done
```

Your method `assertComposerPrerequisitePresent` will then loop over each needed and preform the check. 

To get all the needed extensions, i would suggest to use the latest composer json on master branch and then run a regex for everything with `ext-`

OGKevin reviewed (changes requested):
Some minor refactoring please

Crecket committed:
Check composer file for dependencies and added message constant (bunq/tinker_php#18)

einfachdavid89 commented on setup.sh:
```diff
@@ -5,6 +5,8 @@ ERROR_SYSTEM_UNKNOWN='Unknown system found "%s".\n'
 ERROR_RUN_IN_EMPTY_DIRECTORY='Please run the script from an empty directory\n'
 ERROR_COULD_NOT_FIND_COMMAND='Could not find "%s", try installing it by running "%s".'
 
+MESSAGE_INSTALL_DEPENDENCY='Do you want to try and install this now? (y/N)? '
```
Y

einfachdavid89 replied:
Y
